root/static/scripts/edit/components/GuessCaseOptions.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * @flow
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import expand2react from '../../common/i18n/expand2react';
+import bracketed from '../../common/utility/bracketed';
+import getBooleanCookie from '../../common/utility/getBooleanCookie';
+import setCookie from '../../common/utility/setCookie';
+import * as modes from '../../guess-case/modes';
+import gc from '../../guess-case/MB/GuessCase/Main';
+
+/* eslint-disable flowtype/sort-keys */
+export type ActionT =
+  | {+type: 'set-mode', +mode: string}
+  | {+type: 'set-keep-upper-case', +enabled: boolean}
+  | {+type: 'set-upper-case-roman', +enabled: boolean};
+/* eslint-enable flowtype/sort-keys */
+
+export type DispatchT = (ActionT) => void;
+
+export type StateT = {
+  +keepUpperCase: boolean,
+  +mode: string,
+  +upperCaseRoman: boolean,
+};
+
+export type WritableStateT = {
+  ...StateT,
+};
+
+export type PropsT = $ReadOnly<{
+  ...StateT,
+  +dispatch: (ActionT) => void,
+}>;
+
+export function createInitialState(): StateT {
+  return {
+    keepUpperCase: gc.CFG_UC_UPPERCASED,
+    mode: gc.modeName,
+    upperCaseRoman: getBooleanCookie('guesscase_roman'),
+  };
+}
+
+export function runReducer(
+  state: WritableStateT,
+  action: ActionT,
+): void {
+  switch (action.type) {
+    case 'set-mode': {
+      const modeName = action.mode;
+      gc.modeName = modeName;
+      const mode = modes[modeName];
+      gc.mode = mode;
+      setCookie('guesscase_mode', modeName);
+      state.mode = modeName;
+      break;
+    }
+    case 'set-keep-upper-case': {
+      const enabled = action.enabled;
+      gc.CFG_UC_UPPERCASED = enabled;
+      setCookie('guesscase_keepuppercase', enabled);
+      state.keepUpperCase = enabled;
+      break;
+    }
+    case 'set-upper-case-roman': {
+      const enabled = action.enabled;
+      setCookie('guesscase_roman', enabled);
+      state.upperCaseRoman = enabled;
+      break;
+    }
+  }
+}
+
+const GuessCaseOptions = ({
+  dispatch,
+  keepUpperCase,
+  mode: modeName,
+  upperCaseRoman,
+}: PropsT): React.Element<'div'> => {
+  function handleModeChange(event) {
+    const newModeName = event.target.value;
+
+    if (newModeName !== gc.modeName) {
+      dispatch({mode: newModeName, type: 'set-mode'});
+    }
+  }
+
+  function handleKeepUpperCaseChanged(event) {
+    dispatch({
+      enabled: event.target.checked,
+      type: 'set-keep-upper-case',
+    });
+  }
+
+  function handleUpperCaseRomanChanged(event) {
+    dispatch({
+      enabled: event.target.checked,
+      type: 'set-upper-case-roman',
+    });
+  }
+
+  return (
+    <div id="guesscase-options">
+      <h4>{l('Guess case options')}</h4>
```

Comment:
Why h4 here? The existing dialog uses `.ui-dialog-title` for the title here - is there a specific decision to change the format?

Reply:
The relationship editor popups don't have any title headings, so this probably wouldn't affect those in any case - although we might want to actually add titles to at least some of them, like the batch-add ones :)